### PR TITLE
refactor check functions in sc_src

### PR DESCRIPTION
**Pull request by jze-spyro**

### CODE REVIEW

File: scargo/scargo_src/sc_src.py [hunk 1]
```diff
@@ -482,41 +373,21 @@ def check_clang_tidy(config: Config, verbose: bool) -> None:
...
-            for exclude in exclude_list:
-                if exclude in fname:
-                    logger.info("Skipping %s", fname)
-                    exist_flag = True
-                    break
-            if exist_flag:
+    glob_patterns = (
+        ("*.h", "*.hpp") if headers_only else ("*.h", "*.hpp", "*.c", "*.cpp")
+    )
+    for file_path in chain.from_iterable(
```
Comment (fkuatspyro) [inline scargo/scargo_src/sc_src.py:382]: Doesn't this raise complexity a lot or am I missing something? Maybe os.walk is better option for this kind of use case

**Comment by fkuatspyro** [comment]
Could you write some unit tests for this?

File: scargo/scargo_src/sc_src.py (hunk 1 above)
Comment (jze-spyro) [reply scargo/scargo_src/sc_src.py:382]: Do you mean time or space complexity or some other kind of complexity? https://en.wikipedia.org/wiki/Computational_complexity

Comment (jze-spyro) [reply scargo/scargo_src/sc_src.py:382]: As for time complexity, it does not.

Comment (fkuatspyro) [reply scargo/scargo_src/sc_src.py:382]: Time, I remember doing something like this and benchmarking it eventually and os.walk ended up being a lot faster in my case.

Comment (fkuatspyro) [reply scargo/scargo_src/sc_src.py:382]: Found the example where I benchmarked this. I didn't go into details why this is happening, but I suspect it's creation of Path objects for each file. It's not 1on1 match with your code, but you can take a look:
image

**Comment by jze-spyro** [comment]
> Could you write some unit tests for this?

It's covered by `ut_scargo_check`, isn't it? If you mean `find_files`, it's a helper function, an implementation detail, and as such it should not be tested directly.

File: scargo/scargo_src/sc_src.py (hunk 1 above)
Comment (jze-spyro) [reply scargo/scargo_src/sc_src.py:382]: Ok, maybe it's a bit slower, but don't confuse it with time complexity. It's still the same (linear).

**Comment by fkuatspyro** [comment]
> > Could you write some unit tests for this?
> 
> It's covered by `ut_scargo_check`, isn't it? If you mean `find_files`, it's a helper function, an implementation detail, and as such it should not be tested directly.

Sorry for not being a bit more specific. If you check coverage (actions ->artifacts -> unit-integration-test-report) you will see that some of these functions aren't really tested.

File: scargo/scargo_src/sc_src.py (hunk 1 above)
Comment (jze-spyro) [reply scargo/scargo_src/sc_src.py:382]: I've checked it on my copy of the scargo repository and `os.walk` was only ~2.4 times slower (and we're talking about really negligible times, like less than 0.1 s, so it's no real concern.
On the other hand, with `rglob` I can avoid calling `os.path.join` which was the original goal.

**Review by fkuatspyro — APPROVED** [review]

**Pull request merged**